src/sourabh/Shop.java

```diff
@@ -0,0 +1,100 @@
+package sourabh;
+
+public class Shop {
+	int maggieQty;
+	int dosaQty;
+	int oilPacketsQty;
+	int panipuriQty;
+	int masalaQty;
+	void setInitialStocks(int maggieQtyini, int dosaQtyini, int oilPacketsQtyini, int panipuriQtyini, int masalaQtyini) {
+		maggieQty= maggieQtyini;
+		dosaQty= dosaQtyini;
+		oilPacketsQty= oilPacketsQtyini;
+		panipuriQty= panipuriQtyini;
+		masalaQty= masalaQtyini;
+	}
+
+	void customerPurchase(int maggieQtyCust , int dosaQtyCust, int oilPacketsQtyCust, int panipuriQtyCust, int masalaQtyCust) {
+		System.out.println("*************************");
+		if(maggieQtyCust<= maggieQty) {
+			maggieQty= maggieQty- maggieQtyCust; 
+			System.out.println("Maggie Purchased "+maggieQtyCust);
+		}
+		else{
+			System.out.println("Maggie running out of stock");
+			System.out.println("You can purchase maximum "+maggieQty+" quantity but You requested for "+maggieQtyCust+" quantity.");
+			maggieQty=0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct